packages/vite/src/node/plugins/resolve.ts

```diff
@@ -580,12 +580,21 @@ function tryResolveFile(
   }
 }
 
+export type InternalResolveOptionsWithOverrideConditions =
+  InternalResolveOptions & {
+    /**
+     * @deprecated In future, `conditions` will work like this.
+     * @internal
+     */
+    overrideConditions?: string[]
```

Comment:
What does this do? Hard to know from just reading the code.

Reply:
Ok I think I get it. 

Basically, this option decides which of the default conditions (`production`, `development`, `module`) are allowed.

And it also can add custom conditions, yes?

Reply:
Yes, that's correct.